Lesson18/duygu/script.js

```diff
@@ -0,0 +1,197 @@
+console.log(removeValueFromArray(2, testArrayEx6));
+console.log('----------------------------');
+// 7. Write a function that takes an object and returns the keys as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["name", "age", "city"]
+const testObjEx7 = { name: "John", age: 30, city: "New York" };
+function getObjectKeys(obj) {
+  let keys = [];
+  for (let key in obj) {
+    keys.push(key);
+  }
+  return keys;
+}
+
+console.log("exercise 7:");
+console.log(getObjectKeys(testObjEx7));
+console.log('----------------------------');
+// 8. Write a function that takes an object and returns the values as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["John", 30, "New York"]
+const testObjEx8 = { name: "John", age: 30, city: "New York" };
+function getObjectvalues(obj) {
+  let values = [];
+  for (let key in obj) {
+    values.push(obj[key]);
+  }
+  return values;
+}
+console.log("exercise 8:");
+console.log(getObjectvalues(testObjEx8));
+console.log('----------------------------');
+// 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
+// Input: ({name: "John", age: 30, city: "New York"}, "age")
+// Output: true
+// Input: ({name: "John", age: 30, city: "New York"}, "surname")
+// Output: false
+const testObjEx9 = { name: "John", age: 30, city: "New York" };
+
+function keyExists(obj, key) {
+  return key in obj;
+}
+
+console.log("Ex 9:");
+console.log(keyExists(testObjEx9, "name"));
+console.log(keyExists(testObjEx9, "sity"));
+console.log('----------------------------');
+// 10. Write a function that takes an object and a key-value pair, and adds the key-value pair to the object.
+// Input: ({name: "John", age: 30}, "city", "New York")
+// Output: {name: "John", age: 30, city: "New York"}
+const testObjEx10 = { name: "John", age: 30 };
+
+function addKeyValue(obj, key, value) {
+  obj[key] = value;
+  return obj;
+}
+console.log("Ex 10:");
+console.log(addKeyValue(testObjEx10, "city", "İstanbul"));
+console.log('----------------------------');
+// 11. Write a function that takes an object and a key, and removes the key-value pair from the object.
+// Input: ({name: "John", age: 30, city: "New York"}, "city")
+// Output: {name: "John", age: 30}
+const testObjEx11 = { name: "John", age: 30, city: "New York" };
+
+function removeKeyValue(obj, key) {
+  delete obj[key];
+  return obj;
+}
+console.log("Ex 11:");
+console.log(removeKeyValue(testObjEx11, "city"));
+console.log('----------------------------');
+// 12. Write a function that takes two arrays of equal length and returns an object with the first array as keys and the second array as values.
+// Input: (["name", "age", "city"], ["John", 30, "New York"])
+// Output: {name: "John", age: 30, city: "New York"}
+const array1Ex12 = ["name", "age", "city"];
+const array2Ex12 = ["John", 30, "New York"];
+function addKeysToObject(keys, values) {
```

Comment:
Since this function creates an object from two arrays, better name could be 'createObjectFromArrays(keysArray, valuesArray)`